### ISSUE TITLE
none: exit early with actionable error if there is less RAM than required

### ISSUE THREAD

**Issue by vindhyachalboss** [kubernetes/minikube#10014]
minikube start --driver=none
* minikube v1.16.0 on Ubuntu 20.04
* Using the none driver based on user configuration

X Requested memory allocation (953MB) is less than the recommended minimum 1907M                                                                             B. Deployments may fail.

X The requested memory allocation of 953MiB does not leave room for system overh                                                                             ead (total system memory: 953MiB). You may face stability issues.
* Suggestion: Start minikube with less memory allocated: 'minikube start --memor                                                                             y=953mb'

* Starting control plane node minikube in cluster minikube
* Running on localhost (CPUs=2, Memory=953MB, Disk=7876MB) ...
* OS release is Ubuntu 20.04.1 LTS
* Preparing Kubernetes v1.20.0 on Docker 19.03.8 ...
  - kubelet.resolv-conf=/run/systemd/resolve/resolv.conf
    > kubelet.sha256: 64 B / 64 B [--------------------------] 100.00% ? p/s 0s
    > kubectl.sha256: 64 B / 64 B [--------------------------] 100.00% ? p/s 0s
    > kubeadm.sha256: 64 B / 64 B [--------------------------] 100.00% ? p/s 0s
    > kubeadm: 37.40 MiB / 37.40 MiB [----------------] 100.00% 1.07 GiB p/s 1s
    > kubelet: 108.69 MiB / 108.69 MiB [------------] 100.00% 210.52 MiB p/s 1s
    > kubectl: 38.37 MiB / 38.37 MiB [---------------] 100.00% 61.05 MiB p/s 1s
! initialization failed, will try again: wait: /bin/bash -c "sudo env PATH=/var/                                                                             lib/minikube/binaries/v1.20.0:$PATH kubeadm init --config /var/tmp/minikube/kube                                                                             adm.yaml  --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests,DirAv                                                                             ailable--var-lib-minikube,DirAvailable--var-lib-minikube-etcd,FileAvailable--etc                                                                             -kubernetes-manifests-kube-scheduler.yaml,FileAvailable--etc-kubernetes-manifest                                                                             s-kube-apiserver.yaml,FileAvailable--etc-kubernetes-manifests-kube-controller-ma                                                                             nager.yaml,FileAvailable--etc-kubernetes-manifests-etcd.yaml,Port-10250,Swap": e                                                                             xit status 1
stdout:
[init] Using Kubernetes version: v1.20.0
[preflight] Running pre-flight checks

stderr:
        [WARNING Service-Docker]: docker service is not enabled, please run 'sys                                                                             temctl enable docker.service'
        [WARNING IsDockerSystemdCheck]: detected "cgroupfs" as the Docker cgroup                                                                              driver. The recommended driver is "systemd". Please follow the guide at https:/                                                                             /kubernetes.io/docs/setup/cri/
        [WARNING FileExisting-socat]: socat not found in system path
        [WARNING Service-Kubelet]: kubelet service is not enabled, please run 's                                                                             ystemctl enable kubelet.service'
error execution phase preflight: [preflight] Some fatal errors occurred:
        [ERROR Mem]: the system RAM (953 MB) is less than the minimum 1700 MB
[preflight] If you know what you are doing, you can make a check non-fatal with                                                                              `--ignore-preflight-errors=...`
To see the stack trace of this error execute with --v=5 or higher

*
X Error starting cluster: wait: /bin/bash -c "sudo env PATH=/var/lib/minikube/bi                                                                             naries/v1.20.0:$PATH kubeadm init --config /var/tmp/minikube/kubeadm.yaml  --ign                                                                             ore-preflight-errors=DirAvailable--etc-kubernetes-manifests,DirAvailable--var-li                                                                             b-minikube,DirAvailable--var-lib-minikube-etcd,FileAvailable--etc-kubernetes-man                                                                             ifests-kube-scheduler.yaml,FileAvailable--etc-kubernetes-manifests-kube-apiserve                                                                             r.yaml,FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml,File                                                                             Available--etc-kubernetes-manifests-etcd.yaml,Port-10250,Swap": exit status 1
stdout:
[init] Using Kubernetes version: v1.20.0
[preflight] Running pre-flight checks

stderr:
        [WARNING Service-Docker]: docker service is not enabled, please run 'sys                                                                             temctl enable docker.service'
        [WARNING IsDockerSystemdCheck]: detected "cgroupfs" as the Docker cgroup                                                                              driver. The recommended driver is "systemd". Please follow the guide at https:/                                                                             /kubernetes.io/docs/setup/cri/
        [WARNING FileExisting-socat]: socat not found in system path
        [WARNING Service-Kubelet]: kubelet service is not enabled, please run 's                                                                             ystemctl enable kubelet.service'
error execution phase preflight: [preflight] Some fatal errors occurred:
        [ERROR Mem]: the system RAM (953 MB) is less than the minimum 1700 MB
[preflight] If you know what you are doing, you can make a check non-fatal with                                                                              `--ignore-preflight-errors=...`
To see the stack trace of this error execute with --v=5 or higher

*
* minikube is exiting due to an error. If the above message is not useful, open                                                                              an issue:
  - https://github.com/kubernetes/minikube/issues/new/choose

X Exiting due to GUEST_START: wait: /bin/bash -c "sudo env PATH=/var/lib/minikub                                                                             e/binaries/v1.20.0:$PATH kubeadm init --config /var/tmp/minikube/kubeadm.yaml  -                                                                             -ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests,DirAvailable--va                                                                             r-lib-minikube,DirAvailable--var-lib-minikube-etcd,FileAvailable--etc-kubernetes                                                                             -manifests-kube-scheduler.yaml,FileAvailable--etc-kubernetes-manifests-kube-apis                                                                             erver.yaml,FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml,                                                                             FileAvailable--etc-kubernetes-manifests-etcd.yaml,Port-10250,Swap": exit status                                                                              1
stdout:
[init] Using Kubernetes version: v1.20.0
[preflight] Running pre-flight checks

stderr:
        [WARNING Service-Docker]: docker service is not enabled, please run 'sys                                                                             temctl enable docker.service'
        [WARNING IsDockerSystemdCheck]: detected "cgroupfs" as the Docker cgroup                                                                              driver. The recommended driver is "systemd". Please follow the guide at https:/                                                                             /kubernetes.io/docs/setup/cri/
        [WARNING FileExisting-socat]: socat not found in system path
        [WARNING Service-Kubelet]: kubelet service is not enabled, please run 's                                                                             ystemctl enable kubelet.service'
error execution phase preflight: [preflight] Some fatal errors occurred:
        [ERROR Mem]: the system RAM (953 MB) is less than the minimum 1700 MB
[preflight] If you know what you are doing, you can make a check non-fatal with                                                                              `--ignore-preflight-errors=...`
To see the stack trace of this error execute with --v=5 or higher

*
* If the above advice does not help, please let us know:
  - https://github.com/kubernetes/minikube/issues/new/choose

**Comment by afbjorklund** [comment]
Like it says in the messages, you need "t3.small" or larger.

https://minikube.sigs.k8s.io/docs/start/

> What you’ll need
> * 2 CPUs or more
> * **2GB of free memory**
> * 20GB of free disk space

https://aws.amazon.com/ec2/instance-types/t3/

> Name | vCPUs | Memory (GiB)
> --- | --- | ---
> t3.nano | 2 | 0.5
> t3.micro | 2 | 1.0
> t3.small | 2 | 2.0
> t3.medium | 2 | 4.0

**Comment by afbjorklund** [comment]
The 953M was for the previous k8s versions

**Comment by sharifelgamal** [comment]
Yeah, a t3.micro just doesn't have enough resources to start a minikube cluster properly. The error might look ugly (it's just kubeadm barfing because it can't start up), but the warnings at the beginning of the command are pretty clear. Should make it more clear that we expect failure in this situation? Or should we fail out even?

**Comment by andriyDev** [comment]
Yes this sounds neat!

**Comment by andriyDev** [comment]
As far as I can tell, this seems fixed. My test:
```
$ sudo minikube start --driver=none --memory=953mb
😄  minikube v1.19.0 on Debian rodete
✨  Using the none driver based on user configuration
❗  The 'none' driver does not respect the --memory flag

⛔  Exiting due to RSRC_INSUFFICIENT_REQ_MEMORY: Requested memory allocation 953MiB is less than the usable minimum of 1800MB
```
Also performed the same test on a Ubuntu 20.04 VM (with a system memory limit of 953MB) as well with the same result.

It seemed to be fixed in https://github.com/kubernetes/minikube/commit/a69b41932c9a9ac8658e8d622623e1b211960c71, which seems to be v1.12 though (so probably released in v1.13), so I'm unsure how this error occurred.

Regardless, this issue can be closed.

**Comment by medyagh** [comment]
@vindhyachalboss I am not convinced there is anything we could do since minikube needs min ram for kubenretes
